onnx/src/androidMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/objectdetection/SSDMobileNetObjectDetectionModel.kt

```diff
@@ -0,0 +1,85 @@
+package org.jetbrains.kotlinx.dl.api.inference.onnx.objectdetection
+
+import android.graphics.Bitmap
+import org.jetbrains.kotlinx.dl.api.inference.InferenceModel
+import org.jetbrains.kotlinx.dl.api.inference.objectdetection.DetectedObject
+import org.jetbrains.kotlinx.dl.api.inference.onnx.OnnxInferenceModel
+import org.jetbrains.kotlinx.dl.api.inference.onnx.executionproviders.ExecutionProvider.CPU
+import org.jetbrains.kotlinx.dl.dataset.Coco
+import org.jetbrains.kotlinx.dl.dataset.CocoVersion.V2017
+import org.jetbrains.kotlinx.dl.dataset.preprocessing.*
+import org.jetbrains.kotlinx.dl.dataset.shape.TensorShape
+
+
+private val SSD_MOBILENET_METADATA = SSDModelMetadata(
+    "TFLite_Detection_PostProcess",
+    "TFLite_Detection_PostProcess:1",
+    "TFLite_Detection_PostProcess:2",
+    0, 1
+)
+
+
+public class SSDMobileNetObjectDetectionModel(override val internalModel: OnnxInferenceModel) :
+    SSDObjectDetectionModelBase<Bitmap>(SSD_MOBILENET_METADATA),
+    InferenceModel by internalModel {
+
+    override val classLabels: Map<Int, String> = Coco(V2017, zeroIndexed = true).labels
+
+    private var targetRotation = 0f
+
+    override lateinit var preprocessing: Operation<Bitmap, Pair<FloatArray, TensorShape>>
+        private set
+
+    public constructor (modelBytes: ByteArray) : this(OnnxInferenceModel(modelBytes)) {
+        internalModel.initializeWith(CPU())
+        preprocessing = buildPreprocessingPipeline()
+    }
+
+    // TODO remove code duplication due to different type of class labels array
+    override fun convert(output: Map<String, Any>): List<DetectedObject> {
+        val boxes = (output[metadata.outputBoxesName] as Array<Array<FloatArray>>)[0]
+        val classIndices = (output[metadata.outputClassesName] as Array<FloatArray>)[0]
```

Comment:
Could we figure out which type it is and do a conversion? (maybe not in this pr)

Reply:
Yes, it's possible to get type information using ONNX Api but it will require some work. I will add a task for it.